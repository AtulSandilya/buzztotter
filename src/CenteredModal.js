import { Dimensions, Modal, ScrollView, StyleSheet, Text, TouchableHighlight, View } from 'react-native';
import React, { Component, PropTypes } from 'react';

const styles = StyleSheet.create({
  outerView: {
    backgroundColor: '#222222',
    opacity: 20,
  },
  innerView: {
    backgroundColor: '#ffffff',
    opacity: 100,
  }
});

export default class CenteredModal extends Component {
  static propTypes = {
    isVisible: React.PropTypes.bool,
    closeFromParent: React.PropTypes.func,
    bgOpacity: React.PropTypes.number,
    children: React.PropTypes.element.isRequired,
    heightPercent: React.PropTypes.number,
    widthPercent: React.PropTypes.number,
    animationType: React.PropTypes.string,
  }

  static defaultProps = {
    isVisible: false,
    bgOpacity: 0.6,
    heightPercent: 0.7,
    widthPercent: 0.7,
    animationType: 'fade',
  }

  calcDimensions(dimensionProperty, innerSize){
    let full = Dimensions.get('window')[dimensionProperty]
    return {
      full: full,
      inner: full * innerSize,
      edge: (full - (full * innerSize)) / 2
    }
  }

  render() {
    let width = this.calcDimensions('width', this.props.widthPercent);
    let height = this.calcDimensions('height', this.props.heightPercent);

    return(
      <Modal
        animationType={this.props.animationType}
        transparent={true}
        visible={this.props.isVisible}
        onRequestClose={() => {}}
      >
        <View style={{
          height: height.full,
          width: width.full,
        }}>
          <View style={{
            height: height.full,
            width: width.full,
            backgroundColor: '#222222',
            opacity: this.props.bgOpacity,
            zIndex: 1,
          }}>
            <TouchableHighlight style={{
                height: height.full,
                width: width.full,
              }}
              underlayColor={'#222222'}
              onPress={() => this.props.closeFromParent()}
            >
              <Text> </Text>
            </TouchableHighlight>
          </View>
          <View style={{
            height: height.inner,
            width: width.inner,
            backgroundColor: '#ffffff',
            position: 'absolute',
            top: height.edge,
            left: width.edge,
            zIndex: 100,
          }}>
            <ScrollView style={{zIndex: 100, height: height.inner}}>
              {this.props.children}
            </ScrollView>
          </View>
        </View>
      </Modal>
    );
  }
}
