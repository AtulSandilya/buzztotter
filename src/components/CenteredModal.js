import React, { Component, PropTypes } from 'react';
import { Dimensions, Modal, ScrollView, StyleSheet, Text, TouchableHighlight, View } from 'react-native';

import {globalColors} from './GlobalStyles';

const styles = StyleSheet.create({
  outerView: {
    backgroundColor: '#222222',
    opacity: 20,
  },
  innerView: {
    backgroundColor: '#ffffff',
    opacity: 100,
  }
});

const calcDimensions = (dimensionProperty, innerSize) => {
  let full = Dimensions.get('window')[dimensionProperty]
  return {
    full: full,
    inner: full * innerSize,
    edge: (full - (full * innerSize)) / 2
  }
}

const CenteredModal = ({
  children,
  isVisible = false,
  bgOpacity = 0.6,
  heightPercent = 0.8,
  widthPercent = 0.9,
  animationType = 'fade',
  closeFromParent,
}) => {
  const width = calcDimensions('width', widthPercent);
  const height = calcDimensions('height', heightPercent);
  const cancelButtonSize = 25;

  return(
    <Modal
      animationType={animationType}
      transparent={true}
      visible={isVisible}
      onRequestClose={() => closeFromParent()}
    >
      <View style={{
        height: height.full,
        width: width.full,
      }}>
        <View style={{
          height: height.full,
          width: width.full,
          backgroundColor: 'rgba(34, 34, 34, ' + bgOpacity.toString() + ')',
          zIndex: 1,
        }}>
          <TouchableHighlight style={{
              height: height.full,
              width: width.full,
            }}
            underlayColor={'#222222'}
            onPress={() => closeFromParent()}
          >
          <Text> </Text>
          </TouchableHighlight>
        </View>
        <View style={{
          height: height.inner,
          width: width.inner,
          backgroundColor: '#ffffff',
          position: 'absolute',
          top: height.edge,
          left: width.edge,
          zIndex: 100,
        }}>
          <ScrollView style={{zIndex: 100, height: height.inner}}>
            {children}
          </ScrollView>
        </View>
      </View>
    </Modal>
  );
}

CenteredModal.propTypes = {
  isVisible: React.PropTypes.bool,
  closeFromParent: React.PropTypes.func,
  bgOpacity: React.PropTypes.number,
  children: React.PropTypes.element.isRequired,
  heightPercent: React.PropTypes.number,
  widthPercent: React.PropTypes.number,
  animationType: React.PropTypes.string,
}

export default CenteredModal;
