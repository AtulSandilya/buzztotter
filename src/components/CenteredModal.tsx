import * as React from "react";
import {
  Dimensions,
  Modal,
  ScrollView,
  Text,
  TouchableHighlight,
  View,
} from "react-native";

const calcDimensions = (dimensionProperty, innerSize) => {
  const full = Dimensions.get("window")[dimensionProperty];
  return {
    full: full,
    inner: full * innerSize,
    edge: (full - full * innerSize) / 2,
  };
};

interface CenteredModalProps {
  isVisible: boolean;
  bgOpacity?: number;
  heightPercent?: number;
  widthPercent?: number;
  children?: React.ReactChild;
  closeFromParent?(): void;
}

const CenteredModal: React.StatelessComponent<CenteredModalProps> = ({
  children,
  isVisible = false,
  bgOpacity = 0.6,
  heightPercent = 0.8,
  widthPercent = 0.9,
  closeFromParent,
}) => {
  const width = calcDimensions("width", widthPercent);
  const height = calcDimensions("height", heightPercent);

  return (
    <Modal
      animationType={"fade"}
      transparent={true}
      visible={isVisible}
      onRequestClose={() => closeFromParent()}
    >
      <View
        style={{
          height: height.full,
          width: width.full,
        }}
      >
        <View
          style={{
            height: height.full,
            width: width.full,
            backgroundColor: "rgba(34, 34, 34, " + bgOpacity.toString() + ")",
            zIndex: 1,
          }}
        >
          <TouchableHighlight
            style={{
              height: height.full,
              width: width.full,
            }}
            underlayColor={"#222222"}
            onPress={() => closeFromParent()}
          >
            <Text> </Text>
          </TouchableHighlight>
        </View>
        <View
          style={{
            height: height.inner,
            width: width.inner,
            backgroundColor: "#ffffff",
            position: "absolute",
            top: height.edge,
            left: width.edge,
            zIndex: 100,
          }}
        >
          <ScrollView style={{ zIndex: 100, height: height.inner }}>
            {children}
          </ScrollView>
        </View>
      </View>
    </Modal>
  );
};

export default CenteredModal;
