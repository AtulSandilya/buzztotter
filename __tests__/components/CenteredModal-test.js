import React from 'react';
import { Modal } from 'react-native';

import { shallow } from 'enzyme';

import CenteredModal from '../../build/components/CenteredModal';

jest.mock('react-native-fbsdk', () => '');

describe('CenteredModal component', () => {
  it('renders with no bevegrams', () => {
    const wrapper = shallow(
      <CenteredModal
        isVisible={true}
      />
    )

    expect(wrapper.find(Modal).length).toEqual(1);
  })
})

