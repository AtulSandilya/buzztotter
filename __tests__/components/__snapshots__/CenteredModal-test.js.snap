// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenteredModal component renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(34, 34, 34, 0.6)",
          "height": 1334,
          "width": 750,
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "height": 1334,
              "width": 750,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        >
           
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "backgroundColor": "#ffffff",
            "height": 1067.2,
            "left": 37.5,
            "position": "absolute",
            "top": 133.39999999999998,
            "width": 675,
            "zIndex": 100,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <RCTScrollView
        style={
          Object {
            "borderRadius": 3,
            "height": 1067.2,
            "zIndex": 110,
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
